### ISSUE TITLE
Group anchors are broken

### ISSUE THREAD

**Issue by raphCode** [johannes-wolf/typst-canvas#168]
According to my understanding, using `anchor` on a `group` should place the group content relative to the anchor of the  group bounding box.
However, this does not seems to happen here:
```typst
#import"@preview/cetz:0.1.0"

#cetz.canvas(length:1cm, debug:true, {
  import cetz.draw: *

  group({
    set-style(radius: 0.4)
    circle((0,0))
    circle((1,1))
    circle((2,2))
  }, anchor: "bottom")

  circle((0,0), radius: 0.1, fill: red, stroke: none)
})
```
Result:
![image](https://github.com/johannes-wolf/typst-canvas/assets/15750438/11bc6127-78e2-4cf7-9aa8-c781b47fcd4c)

If this is indeef the intended behavior, I do not understand what is going on. In this case, the docs might need some clarification.

**Comment by fenjalien** [comment]
I believe this is intended behaviour. I'm struggling to find the right words to define it but `anchor` states which anchor the element should be positioned relative to. Another way of putting it would be which anchor should be positioned on the given coordinate, by default this is the `center` anchor.

Looking at the docs for it, the example does make sense but it could be explained better:
![image](https://github.com/johannes-wolf/typst-canvas/assets/34489450/43838982-2c3c-4574-8356-60773dcbd74a)
The large circle is given the coordinate `(0,0)` and `anchor` "left", so its left anchor is placed at `(0,0)` while its center anchor is at `(1,0)`. In your example, the bottom anchor of the group element has been placed where its center anchor would be.

Does this make sense? Could you also provide an image of the result you expected? It would help to see where the confusion is...

**Comment by johannes-wolf** [comment]
This is expected behavior as the default anchor is "center". So translating the group by ("bottom" - "center") gives this result. I think groups and anchors are somewhat confusing because you do not specify a groups position in the first place, but it gets positioned with its children.

**Comment by raphCode** [comment]
> Another way of putting it would be which anchor should be positioned on the given coordinate, by default this is the `center` anchor.

Yes, this is how I understood anchors. The example from the docs you showed makes perfect sense to me: The circle's anchor is placed at the given coordinates.

> I think groups and anchor positioning are somewhat confusing because you do not specify a groups position in the first place, but it gets positioned by its children.

Yes, this is where my confusion comes from. The intuition from the circle anchor example does not directly extend to groups.

My use case would have been to vertically align the centers of multiple groups. I was surprised that `anchor: "center"` did not work, but `anchor: "top"` did. Judging from your explanations, the correct anchor also depends on the positions of the children.

Would it be worthwhile to add an optional position parameter to the `group` command that overrides the group placement?

**Comment by johannes-wolf** [comment]
Position as in coordinates? You can already do that by calling `set-origin` or `translate`.

**Comment by raphCode** [comment]
Yes, kind of.
Maybe I am not seeing a more obvious solution, but consider my use case:
I want to draw a small icon, say something that resembles a database, like this:
![image](https://github.com/johannes-wolf/typst-canvas/assets/15750438/44ca0fd5-7bb3-4b79-a7cd-d19ef5960c53)

 I now want to reuse this drawing of the icon at multiple locations (and add text, arrows, etc). For example, I want to place the icon's center at (2, 2).
```typst
#import"@preview/cetz:0.1.1"

#set page(width: auto, height: auto)

#cetz.canvas(length:1cm, {
  import cetz.draw: *

  let database(pos) = group({
    set-origin(pos)
    scale((y: 0.5))
    let n = 3
    let r = 2
    set-origin((0, -n / 2))
    for y in range(n) {
      arc((r, y), start: 0deg, stop: -180deg, radius: r)
    }
    circle((0, n), radius: r)
    line((r, 0), (r, n))
    line((-r, 0), (-r, n))
  })

  database((2, 2))

  circle((0, 0), radius: 0.1, fill: red, stroke: none)
  circle((2, 2), radius: 0.1, fill: blue, stroke: none)
})
```
![image](https://github.com/johannes-wolf/typst-canvas/assets/15750438/efe0ed5e-581c-4a49-aca3-cebb1f3798e7)

I have to do multiple steps to make it work:
- `set-origin((0, -n / 2))`: to center the drawing inside the group. In other words, make the group's center anchor the center of the icon(s bounding box).
- add a `pos` argument so I can call `set-origin(pos)` inside the group

I realize what I probably really want is an alternative to the current implementation of the `group` command. It would behave like this:
I can draw freely inside the group without having to pay attention where the origin is. Afterwards the group calculates a bounding box around the drawing with the default anchors. I can then reuse the group and place its center (and thus the center of the icon) at a certain position.
Since this already spiraled off into a feature request: It would be really great if the group could add some kind of padding (inset / outset) to the bounding box, so arrows can be attached with a little space.

I realize that some (all?) of these features can probably already be implemented with `set-origin` and `anchor`, but It would be very nice if cetz provided them.

**Comment by johannes-wolf** [comment]
We could just add a `position: <coordinate>` to `group(..)` that would translate the groups content by `position - group.anchor`. @fenjalien?

**Comment by johannes-wolf** [comment]
But I think the current way of doing it `set-origin()` is pretty easy.
You just have to draw everything in your group with origin (0,0,0)!

```typst
#box(stroke: 2pt + red, canvas(length: 1cm, {
  import draw: *

  let face(position) = group({
    circle(position, fill: red, radius: .1)
    set-origin(position)
    circle((0, 0))
    arc((0,0), start: -180deg, stop: 0deg, anchor: "origin", radius: .7)
    circle((.3, .2), radius: (.2, .3))
    circle((-.3, .2), radius: (.2, .3))
    circle((-.23, .1), radius: .1, fill: black)
    circle(( .37, .1), radius: .1, fill: black)
  }, anchor: "center")

  face((0,0))
  face((3,0))
}))
```

![grafik](https://github.com/johannes-wolf/typst-canvas/assets/519002/5cad5f4c-0d06-42b8-86fb-b18c2887ca64)

**Comment by raphCode** [comment]
Okay, this works for me.
Just for the record, here is my function to create anchors with a margin:

```
let margin(ct, name, x: 0.2) = {
  let a(n, offset) = anchor(n, (rel: offset, to: "g." + n))
  group({
    group(ct, name: "g")
    a("left", (x: -x))
    a("right", (x: x))
    a("bottom", (y: -x))
    a("top", (y: x))
    a("bottom-left", (-x, -x))
    a("bottom-right", (x, -x))
    a("top-left", (-x, x))
    a("top-right", (x, x))
  }, name: name)
}

margin(circle(()), "c")
```

![image](https://github.com/johannes-wolf/typst-canvas/assets/15750438/768762ae-4fb8-4827-80d9-41a011baec87)
